scalafix-core/shared/src/main/scala/scalafix/internal/patch/EscapeHatch.scala

```diff
@@ -302,38 +300,39 @@ object EscapeHatch {
       var currentlyDisabledRules = Map.empty[String, Position]
 
       def enable(
-          offset: EscapeOffset,
-          anchor: Token.Comment,
-          rules: String): Unit =
-        enableBuilder += (offset -> filtersFor(offset, anchor, rules))
+          rules: String,
+          start: EscapeOffset,
+          anchor: Token.Comment): Unit =
+        enableBuilder += (start -> makeFilters(start, None, anchor, rules))
 
       def disable(
-          offset: EscapeOffset,
-          anchor: Token.Comment,
-          rules: String): Unit =
-        disableBuilder += (offset -> filtersFor(offset, anchor, rules))
-
-      def filtersFor(
-          offset: EscapeOffset,
+          rules: String,
+          start: EscapeOffset,
+          end: Option[EscapeOffset],
+          anchor: Token.Comment): Unit =
+        disableBuilder += (start -> makeFilters(start, end, anchor, rules))
+
+      def makeFilters(
+          start: EscapeOffset,
+          end: Option[EscapeOffset],
           anchor: Token.Comment,
           rules: String): List[EscapeFilter] = {
-        val splittedRules = splitRules(rules)
+        val splitRules0 = splitRules(rules)
 
-        if (splittedRules.isEmpty) { // wildcard
-          List(EscapeFilter(FilterMatcher.matchEverything, anchor.pos, offset))
+        if (splitRules0.isEmpty) { // wildcard
+          EscapeFilter(FilterMatcher.matchEverything, anchor.pos, start, end) :: Nil
```

Comment:
I prefer `List(x)` over `x :: Nil` it makes it easier to see the return type.

Reply:
👍